File: App/src/Adaptive.ReactiveTrader.Client/Transport/IConnection.cs
using System;
using System.Reactive;
using Microsoft.AspNet.SignalR.Client;

namespace Adaptive.ReactiveTrader.Client.Transport
{
    public interface IConnection
    {
        IObservable<ConnectionStatus> Status { get; }
        IObservable<bool> IsConnected { get; }
        IObservable<Unit> Initialize();
        IHubProxy ReferenceDataHubProxy { get; }
        IHubProxy PricingHubProxy { get; }
        IHubProxy ExecutionHubProxy { get; }
        IHubProxy BlotterHubProxy { get; }
    }
 }

File: App/src/Adaptive.ReactiveTrader.Client.JS/Transport/IConnection.ts
/// <reference path="../typings/signalr/signalr.d.ts"/> 
/// <reference path="../typings/rx.js/rx.d.ts"/>

interface IConnection {
    status: Rx.Observable<ConnectionInfo>;
    initialize(): Rx.Observable<{}>;
    address: string;
    referenceDataHubProxy: HubProxy;
    pricingHubProxy: HubProxy;
    executionHubProxy: HubProxy;
    blotterHubProxy: HubProxy;
}